posthog/temporal/workflows/squash_person_overrides.py

```diff
@@ -0,0 +1,698 @@
+import json
+import os
+from collections.abc import Iterator
+from dataclasses import asdict, dataclass, field
+from datetime import datetime, timedelta, timezone
+from typing import Iterable, NamedTuple
+from uuid import UUID
+
+import psycopg2
+from django.conf import settings
+from temporalio import activity, workflow
+from temporalio.common import RetryPolicy
+
+from posthog.clickhouse.client.execute import sync_execute
+from posthog.temporal.workflows.base import CommandableWorkflow
+
+EPOCH = datetime(1970, 1, 1, 0, 0, tzinfo=timezone.utc)
+
+SELECT_PERSONS_TO_DELETE_QUERY = """
+SELECT
+    team_id,
+    old_person_id,
+    override_person_id,
+    max(created_at) AS latest_created_at,
+    max(version) AS latest_version,
+    min(oldest_event) AS oldest_event_at
+FROM
+    {database}.person_overrides
+WHERE
+    created_at <= %(latest_created_at)s
+GROUP BY
+    team_id, old_person_id, override_person_id
+"""
+
+SELECT_LATEST_CREATED_AT_QUERY = """
+SELECT
+    max(created_at)
+FROM {database}.person_overrides;
+"""
+
+CREATE_DICTIONARY_TABLE_QUERY = """
```

Comment:
nit: maybe `DICTIONARY_SOURCE_TABLE` is more clear

Reply:
With the new approach we are using the ReplacingMergeTree table `person_overrides`, so this query doesn't exist anymore.